src/io/iohk/scalanet/peergroup/PeerGroup.scala

```diff
@@ -21,4 +21,24 @@ object PeerGroup {
   trait TerminalPeerGroup[A, F[_]] extends PeerGroup[A, F]
 
   case class InitializationError(message: String, cause: Throwable)
+
+  def create[PG](pg: => PG, config: Any): Either[InitializationError, PG] =
```

Comment:
I wonder if it would be better to receive `pg: => Try[PG]` instead to make it obvious that an exception can be thrown.

Why is the config required if the peer group can be created without it? If you keep it, please add a type parameter for the config, `Any` is just too general.